pymc_marketing/mlflow.py

```diff
@@ -412,6 +413,189 @@ def log_inference_data(
+        Additional keyword arguments to pass to the selected sampling methods.
+
+    Examples
+    --------
+    MLFlow Registering for a PyMC-Marketing model:
+
+    .. code-block:: python
+
+        import pandas as pd
+
+        import mlflow
+
+        from pymc_marketing.mmm import (
+            GeometricAdstock,
+            LogisticSaturation,
+            MMM,
+        )
+        import pymc_marketing.mlflow
+        from pymc_marketing.mlflow import MMMRegistrar
+
+        pymc_marketing.mlflow.autolog(log_mmm=True)
+
+        # Usual PyMC-Marketing model code
+
+        data_url = "https://raw.githubusercontent.com/pymc-labs/pymc-marketing/main/data/mmm_example.csv"
+        data = pd.read_csv(data_url, parse_dates=["date_week"])
+
+        X = data.drop("y",axis=1)
+        y = data["y"]
+
+        mmm = MMM(
+            adstock=GeometricAdstock(l_max=8),
+            saturation=LogisticSaturation(),
+            date_column="date_week",
+            channel_columns=["x1", "x2"],
+            control_columns=[
+                "event_1",
+                "event_2",
+                "t",
+            ],
+            yearly_seasonality=2,
+        )
+
+        # Incorporate into MLflow workflow
+        mlflow_mmm = MMMRegistrar(
+            mmm,
+            extend_idata=False,
+            combined=True,
+            include_last_observations=False,
+            original_scale=True,
+        )
+        model_name = "my_mmm_model"
+
+        mlflow.set_experiment("MMM Experiment")
+
+        with mlflow.start_run():
+            idata = mmm.fit(X, y)
+
+            # Additional specific logging
+            fig = mmm.plot_components_contributions()
+            mlflow.log_figure(fig, "components.png")
+
+            # Register the model
+            mlflow.pyfunc.log_model(
+                artifact_path=model_name,
+                python_model=mlflow_mmm,
+            )
+            run_id = mlflow.active_run().info.run_id
+            model_uri = f"runs:/{run_id}/{model_name}"
+            mlflow.register_model(model_uri, model_name)
+
+    """
+
+    def __init__(
+        self,
+        model: MMM,
+        predict_method: str | None = "predict",
+        extend_idata: bool = False,
+        combined: bool = True,
+        include_last_observations: bool = False,
+        original_scale: bool = True,
+        var_names: list[str] | None = None,
+        **sample_kwargs: dict,
+    ):
+        self.model = model
+        self.predict_method = predict_method
+        self.extend_idata = extend_idata
+        self.combined = combined
+        self.include_last_observations = include_last_observations
+        self.original_scale = original_scale
+        self.var_names = (
+            var_names if var_names is not None else ["y"]
+        )  # Initialize if not provided
+        self.sample_kwargs = sample_kwargs
+
+    def predict(
+        self, context: Any, model_input, params: dict[str, Any] | None = None
```

Comment:
Yes exactly!

Unfortunately I don't think you can have a whole lot of control over the params you pass. To conform with their API I believe you can only pass in a choice of prediction method - other args won't work. I tried to get around this by having the user optionally pass in sampler_kwargs at time of class instantiation (at least then the prediction methods are set somewhat flexibly)


[Docs](https://mlflow.org/docs/latest/traditional-ml/creating-custom-pyfunc/notebooks/override-predict.html)
> We can only provide a params override for the key predict_method. Anything apart from this will be ignored and a warning will be shown indicating that the unknown parameter will not be passed to the underlying model.

Reply:
@wd60622  hopefully my answer was satisfactory? I'll address your other outstanding feedback on the [new PR](https://github.com/pymc-labs/pymc-marketing/pull/1026) but thought it made sense to close this one here ☺️